<launch>
<!-- ============ Arguments ============ -->
  <!-- Groundtruth -->
  <arg name="ground_truth_pointcloud_file" default="/home/lukas/Documents/Datasets/flat_dataset/ground_truth/run1/flat_1_gt_10000.ply"/>
  
  <!-- Source map -->
  <arg name="map_file" default="/home/lukas/Documents/PanopticMapping/Data/evaluations/fixed_resolution/10cm.panmap"/>
  
  <!-- Params -->
  <arg name="maximum_distance" default="100000"/>
  <arg name="evaluate" default="true"/>
  <arg name="compute_coloring" default="false"/>
  <arg name="color_by_max_error" default="false"/>
  <arg name="visualize" default="false"/>
  <arg name="ignore_truncated_points" default="true"/>

<!-- ============ Evaluations ============ -->
  <node name="evaluator" pkg="panoptic_mapping_utils" type="single_map_evaluation" output="screen" required="true">
    <param name="map_file" value="$(arg map_file)" /> 
    <param name="ground_truth_pointcloud_file" value="$(arg ground_truth_pointcloud_file)" /> 
    <param name="maximum_distance" value="$(arg maximum_distance)" />     
    <param name="visualize" value="$(arg visualize)" /> 
    <param name="evaluate" value="$(arg evaluate)" /> 
    <param name="compute_coloring" value="$(arg compute_coloring)" /> 
    <param name="color_by_max_error" value="$(arg color_by_max_error)" /> 
    <param name="ignore_truncated_points" value="$(arg ignore_truncated_points)" /> 
  </node>
  
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find panoptic_mapping_utils)/config/evaluate_panmap.rviz" if="$(arg visualize)"/>
  
</launch>
